csharp/extractor/Semmle.Extraction.CSharp/Populators/CompilationUnit.cs

```diff
@@ -57,10 +58,16 @@ public override void VisitAttributeList(AttributeListSyntax node)
                 return;
 
             var outputAssembly = Assembly.CreateOutputAssembly(cx);
+            var attributeDatas = cx.Compilation.Assembly.GetAttributes().ToList();
+            attributeDatas.AddRange(cx.Compilation.Assembly.Modules.SelectMany(m => m.GetAttributes()));
             foreach (var attribute in node.Attributes)
             {
-                var ae = new Attribute(cx, attribute, outputAssembly);
-                cx.BindComments(ae, attribute.GetLocation());
+                var attributeData = attributeDatas.Single(ad => ad.ApplicationSyntaxReference?.GetSyntax() == attribute);
+                if (attributeData is object)
+                {
+                    var ae = Attribute.Create(cx, attributeData, outputAssembly);
+                    cx.BindComments(ae, attribute.GetLocation());
+                }
```

Comment:
This is quite ugly, but I couldn't find a better way to extract assembly attributes.